src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/alignment/AssemblyContigAlignmentsConfigPicker.java

```diff
@@ -571,13 +572,30 @@ private static double computeTigExplainQualOfOneConfiguration(final List<Alignme
             return bestConfigurations.stream()
                     .map(mappings -> splitGaps(mappings, false))
                     .map(mappings -> removeNonUniqueMappings(mappings, ALIGNMENT_MQ_THRESHOLD, ALIGNMENT_LOW_READ_UNIQUENESS_THRESHOLD))
+                    .filter(mappings ->  {
+                        if (mappings.goodMappings.size() != 2)
+                            return true;
+                        else {
+
+                            if ( simpleChimeraWithStichableAlignments(mappings.goodMappings.get(0), mappings.goodMappings.get(1)) )
+                                return false;
+                            else
+                                return true;
+                        }
+                    })
                     .map(mappings -> createContigGivenClassifiedAlignments(contigName, contigSeq, mappings, true))
                     .sorted(getConfigurationComparator())
                     .iterator();
         } else {
             final GoodAndBadMappings intermediate = splitGaps(bestConfigurations.get(0), false);
             final GoodAndBadMappings result = removeNonUniqueMappings(intermediate, ALIGNMENT_MQ_THRESHOLD, ALIGNMENT_LOW_READ_UNIQUENESS_THRESHOLD);
-            return Collections.singletonList(createContigGivenClassifiedAlignments(contigName, contigSeq, result, false)).iterator();
+            if (result.goodMappings.size() != 2
```

Comment:
If you do the refactoring suggested above you could make a little method for the if clause and only write it once.

Reply:
done